pkg/controllers/datastoresyncer/node_handler.go

```diff
@@ -35,7 +37,21 @@ func (d *DatastoreSyncer) handleCreateOrUpdateNode(obj runtime.Object, numRequeu
 
 	globalIPOfNode := node.GetAnnotations()[constants.SmGlobalIP]
 
-	return d.updateLocalEndpointIfNecessary(globalIPOfNode)
+	// Validate that globalIPOfNode falls in the globalCIDR allocated to the cluster.
+	if globalIPOfNode != "" {
+		_, ipnet, err := net.ParseCIDR(d.localCluster.Spec.GlobalCIDR[0])
+		if err != nil {
+			// Ideally this will not happen as globalCIDR is expected to be a valid CIDR.
+			klog.Errorf("Error parsing the GlobalCIDR %q: %v", d.localCluster.Spec.GlobalCIDR, err)
+			return false
+		}
+
+		if ipnet.Contains(net.ParseIP(globalIPOfNode)) {
+			return d.updateLocalEndpointIfNecessary(globalIPOfNode)
+		}
```

Comment:
What would be the scenario where the global IP wouldn't be valid wrt to the CIDR? Perhaps we should log an error or warning?

Reply:
Basically I noticed this issue while debugging an issue where the same cluster was used for multiple re-deployments. Globalnet adds a globalIP annotation to the Gateway node which is used as healthcheck-ip.
Ideally this annotation should be removed as part of `subctl uninstall ....` operation, but it was not (will push a separate PR to fix it), so its creating issue during re-deployment when a different GlobalCIDR was allocated to the same cluster. 

Please see - https://github.com/submariner-io/submariner/issues/1780#issuecomment-1093293544